### PR TITLE
Update UG to resolve bugs from PE-D

### LINKED ISSUE #233: Bug: User is allowed to add addition parameters to a command if the parameter is unused.
Something like this is allowed when not intended:

* `addc c/CS1231S s/Lab3`
* `help n/ME`

### LINKED ISSUE #277: [PE-D][Tester A] Combine the 2 command tables into 1
At first glance of the UG, I thought there was a missing command table for the features in focus mode as the command table was placed further down. Maybe can combine both command tables together or put them side by side.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: Dilysss/ped#4

### LINKED ISSUE #285: [PE-D][Tester D] No warnings raised when unassigning students when not from the class
Students can be unassigned from a class which they aren't in without an error or warning. Users might accidentally key in the index wrongly, so this might be an issue

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: XenonShawn/ped#6

### LINKED ISSUE #294: [PE-D][Tester A] Student can be assigned to the same class multiple times
Although I have already added a student into a class. I can still input the assign command to add them in again. There is no error message stating that the student already exist

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Dilysss/ped#9

### LINKED ISSUE #300: [PE-D][Tester D] Repetition of Documentation
There are a few instances of documentation repetition throughout the UG, one of the more obvious examples is as follows. It may be good to either provide more information (or in the case of the clear command, a warning).

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: XenonShawn/ped#13

### LINKED ISSUE #307: [PE-D][Tester B] Not displaying *all* session grades
As seen from the 2nd photo, Johns grade for lab1 is not displayed, when the user guide said it should

Think this should be an easy fix, one can simply edit the UG to say it is only for sessions that have been graded~

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: KJunWei/ped#3

### PR DESCRIPTION
**Pull request by Bubbl3T**
Resolves #233, #277, #285, #294, #300, #307.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -47,34 +47,39 @@ If you encounter any issues in launching and using the app, feel free refer to t
...
 
-### Modes
+## Modes
 In TA-Assist, you can switch into a mode called the **focus** mode, which lets you run tasks that are specific to (module) class. Therefore,
 [some commands](#features-available-in-focus-mode) can only be executed when you are in focus mode. Commands that you can run in the default (unfocused) mode can also
 be run in focus mode. On the other hand, commands that are available only in focus mode cannot be executed in the default (unfocused) mode.
 
 Let's first begin with the commands available in the default mode.
 
+## Commands available in default mode
```
Comment (RezwanArefin01) [inline docs/UserGuide.md]: There was another (closed) issue that got confused here. Commands that are available in the beginning are also available in focus mode. So terming it "commands available in default mode" and "commands available in focus mode" may be a bit confusing. Can you find a way to write this better? 

Also "default mode" is a new term introduced here without prior explanation.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -194,10 +199,10 @@ Format: `addc c/CLASS_NAME...`
 * Add classes with specified names. 
 * The class names are **case-insensitive**.
   * e.g. If a class with a name **CS1101S** already exists, `addc c/cs1101s` does not add a
-  new class `cs1101s`.
+    new class `cs1101s`.
 * If there are duplicate class names, the class name is taken to be the last valid class parameter `c/`.
   * e.g. If a class with a name **CS1101S** does not exist, `addc c/CS1101S c/cs1101s` adds one class with the name
-  **cs1101s**.
+    **cs1101s**.
```
Comment (RezwanArefin01) [inline docs/UserGuide.md]: One of cs1101s is in monospace, another in bold. Is there a reason for the difference?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -294,22 +301,23 @@ Clears all existing data in TA-Assist.
...
-| `scores`  | `scores s/SESSION_NAME`                       |
-| `view`    | `view INDEX`                                  |
-| `export`  | `export`                                      |
-| `unfocus` | `unfocus`                                     |
+| `scores`  | `scores s/SESSION_NAME`                  |
+| `view`    | `view INDEX`                             |
+| `export`  | `export`                                 |
+| `unfocus` | `unfocus`                                |
+
+*Click [here](# Commands available in default mode) for commands available in default mode.*
```
Comment (RezwanArefin01) [inline docs/UserGuide.md]: Maybe mention here that the commands available in default mode is also available in focus mode. See if there are exceptions; then mention them too.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -420,7 +428,8 @@ Views all session grades of a student within the focused class.
 " %}
 
 Format: `view INDEX`
-* Views the grade of the student at index `INDEX` for the currently focused class.
+* Views the grades of the student at index `INDEX` for the currently focused class.
```
Comment (RezwanArefin01) [inline docs/UserGuide.md]: "Views" is probably not the right word here. Maybe "Shows", "Displays"?

**Review by RezwanArefin01 — CHANGES_REQUESTED** [review]
Suggested minor changes for more clarification.

File: docs/UserGuide.md (hunk 1 above)
Comment (Bubbl3T) [reply docs/UserGuide.md]: I think the first list can be commands for TA-Assist, then the second list can be commands ONLY available in focus mode, then we perhaps put a note that the original list of commands are also available in focus mode.

File: docs/UserGuide.md (hunk 2 above)
Comment (Bubbl3T) [reply docs/UserGuide.md]: Good spot, this should not be the case.

File: docs/UserGuide.md (hunk 4 above)
Comment (Bubbl3T) [reply docs/UserGuide.md]: That sounds like a good choice!

**Review by RezwanArefin01 — APPROVED** [review]
Looks good to merge!

**Pull request merged**